pyiron_base/project/maintenance.py

```diff
@@ -102,6 +139,67 @@ def defragment_storage(
             hdf = job.project_hdf5
             hdf.rewrite_hdf5(job.name)
 
+    def update_hdf_types(
+        self,
+        recursive: bool = True,
+        progress: bool = True,
+        **kwargs: dict,
+    ):
+        """
+        Rewrite TYPE fields in hdf5 files for renamed modules.
```

Comment:
Should `_MODULE_CONVERSION_DICT` be an attribute of the maintenance class? It feels like here we should direct the reader to where these "renamed modules" are, and that hint makes more sense to me if they also just live on this class.

Reply:
I had it as global, because I didn't want someone to accidentally mess with it.  Definitely this docstring should reference `add_module_conversion`.  If `_MODULE_CONVERSION_DICT` becomes an attribute, `add_module_conversion` would have to become a method.  Feels a bit weird, because the other methods are all for actual fixes, whereas that one only sets some fixes up.  Dunno, will mull over it on the train tomorrow.

Reply:
I've left it as is for now, but added comments.  Having it as a global has the advantage that we only need to maintain one such dictionary, instead of carrying it around on each project, which might make it easy to loose it when we define the additional Project sub classes.